connector/src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala

```diff
@@ -158,12 +165,17 @@ case class BigQueryWriteHelper(bigQuery: BigQuery,
 
   def friendlyTableName: String = BigQueryUtil.friendlyTableName(options.tableId)
 
-  def cleanTempraryGcsPath: Unit = {
+  def cleanTemporaryGcsPathIfNeeded: Unit = {
     // TODO(davidrab): add flag to disable the deletion?
-    createTemporaryPathDeleter.deletePath
+    createTemporaryPathDeleter.map(_.deletePath)
   }
 
-  private def createTemporaryPathDeleter = IntermediateDataCleaner(temporaryGcsPath, conf)
+  private def createTemporaryPathDeleter: Option[IntermediateDataCleaner] = {
```

Comment:
Great idea to use an `Option` here, but its instantiation can be simpler: 
```
private def createTemporaryPathDeleter: Option[IntermediateDataCleaner] = options.temporaryGcsBucket.map(IntermediateDataCleaner(gcsPath, conf))
```

Reply:
Done!